pkg/skaffold/build/local/jib.go

```diff
@@ -18,16 +18,41 @@ package local
 
 import (
 	"context"
+	"crypto/sha1"
+	"encoding/hex"
 	"io"
+	"os/exec"
+	"regexp"
 
-	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3"
-	"github.com/pkg/errors"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/sirupsen/logrus"
 )
 
-func (b *Builder) buildJibMaven(_ /*ctx*/ context.Context, _ /*out*/ io.Writer, _ /*workspace*/ string, _ /*a*/ *v1alpha3.JibMavenArtifact) (string, error) {
-	return "", errors.New("buildJibMaven is unimplemented")
+// executeBuildCommand executes the command-line with the working directory set to `workspace`.
+func executeBuildCommand(ctx context.Context, out io.Writer, workspace string, commandLine []string) error {
+	logrus.Infof("Building %v: %v", workspace, commandLine)
+	cmd := exec.CommandContext(ctx, commandLine[0], commandLine[1:]...)
+	cmd.Dir = workspace
+	cmd.Stdout = out
+	cmd.Stderr = out
+	return util.RunCmd(cmd)
 }
 
-func (b *Builder) buildJibGradle(_ /*ctx*/ context.Context, _ /*out*/ io.Writer, _ /*workspace*/ string, _ /*a*/ *v1alpha3.JibGradleArtifact) (string, error) {
-	return "", errors.New("buildJibGradle is unimplemented")
+// jibBuildImageRef generates a valid image name for the workspace and project.
+// The image name is always prefixed with `jib`.
+func generateJibImageRef(workspace string, project string) string {
+	imageName := "jib" + workspace
+	if project != "" {
+		imageName += "_" + project
+	}
+	// if the workspace + project is a valid image name then use it
+	match := regexp.MustCompile(constants.RepositoryComponentRegex).MatchString(imageName)
+	if match {
```

Comment:
if `match` is not used outside of the `if`, then it's a good practice in Go to write:

```
if match := regexp.MustCompile(constants.RepositoryComponentRegex).MatchString(imageName); match {
  return imageName
}
```

Of course this can also be written as:

```
if regexp.MustCompile(constants.RepositoryComponentRegex).MatchString(imageName) {
  return imageName
}```

Reply:
(This is a Nit)